specification.md

```diff
@@ -1059,6 +1061,65 @@ Note that the referenced function definition type in this case must be `rest` (d
 
 For more information about functions, reference the [Functions definitions](#Function-Definition) section.
 
+#### Using functions for HTTP Service Invocations
+
+For a few reasons, workflow designers might see necessary making straight HTTP invocations to other systems, such as:
+
+1. Target service might not expose an OpenAPI specification descriptor
+2. Target service might not implement RESTful interfaces
+3. The use case requires a simple HTTP request invocation
+
+The specification supports making HTTP invocations via a simplified interface in the [functions definition](#Function-Definition).
+
+The HTTP invocation interface can be described in the `operation` attribute of the function definition, as shown in the example:
+
+```json
+{
+"functions": [
+  {
+    "name": "queryUserById",
+    "operation": "get:/users/{id}"
+  }
+]
+}
+```
+
+In this case, the `operation` attribute is composed by:
+
+* The HTTP method used by the HTTP invocation. Can be any [valid HTTP method](https://www.rfc-editor.org/rfc/rfc9110.html#name-method-definitions) such as `get`, `post`, `put`, etc.
+* The endpoint location path. Runtimes implementations are **discouraged** to describe the full URL in this field since this is a matter of infrastructure configuration.
+
+The table below lists how to describe the HTTP parameters in the endpoint location path:
+
+| Parameter | Description | Example |
+| --- | --- | --- |
+| Path | Path parameters are part of the location and must be enclosed with brackets (`{}`). The parameter name can be later referenced by the workflow states. | `/user/{id}` |
+| Query | [Query parameters](https://www.rfc-editor.org/rfc/rfc9110.html#section-4.2.2) must be enclosed with brackets (`{}`) after the query key separator (`=`). The parameter name can be later referenced by the workflow states. | `/user?status={status}` |
+
+Additional HTTP Headers can be passed via the `metadata` function definition. For example:
+
+```json
+{
+"functions": [
+  {
+    "name": "queryUserById",
+    "operation": "get:/users/{id}",
+    "metadata": {
```

Comment:
wouldnt put metadata in spec doc, maybe if you want to showcase this is better to add it to a new sample

Reply:
In this case, HTTP Headers are metadata of the request in a sense. What we can do is:

1. Not to describe headers in the operation or function definition and encourage implementations to add a way to do it via configurations. We can argue that's an infra use case.
2. Let the metadata solve this problem like we are doing
3. Introduce an extension for this case, so implementations can optionally use them, so users could define their custom headers there.

I don't have a strong opinion, but I like the idea to describe headers params using the `metadata` field: https://www.rfc-editor.org/rfc/rfc9110.html#section-6

> a headers lookup table of name/value pairs for extending that control data and conveying additional information about the sender, message, content, or context

Reply:
ok but see this from user perspective, spec has now a built in feature for http 
but your core example shows the use of metadata to get it working

im not saying not to like use metadata, but for core spec definition http invocation 
metadata should not be necessary.

Reply:
imo all we need to do here is add the most basic way to do a get/post
nothing more nothing else, make samples as simple as possible please

Reply:
ok lets discuss on zoom so u can explain me pls